### PR TITLE
added editorconfig file

### LINKED ISSUE #26: Adding .editorconfig file.
it would be good to have .editorconfig file to maintain indentation uniformity.
@iiitv/project-maintainers if this issue is valid i would like to work on it.

### PR DESCRIPTION
**Pull request by nlok5923**
fixes : #26

### CODE REVIEW

**Comment by nlok5923** [comment]
@iiitv/project-maintainers sir please review this pull request.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+[*]
+trim_trailing_whitespace = true
+end_of_line = lf
+charset = utf-8
+
+[*.{js,css}]
+indent_style = space
+indent_size = 2
+insert_final_newline = true
```
Comment (aashutoshrathi) [inline .editorconfig]: Move this to `[*]`

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+max_line_length = 80
+
+[*.html]
+indent_style = space
+indent_size = 4
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (aashutoshrathi) [inline .editorconfig]: Always add a newline to end.

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]

**Comment by aashutoshrathi** [comment]
I meant move `insert_final_newline = true` to `[*]`

File: .editorconfig [hunk 3]
```diff
@@ -4,15 +4,17 @@ root = true
...
 
 [*.{js,css}]
+indent_style = space
+indent_size = 2
 max_line_length = 80
 
 [*.html]
+indent_style = space
 indent_size = 4
+insert_final_newline = true
```
Comment (aashutoshrathi) [inline .editorconfig]: No need here now.

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]
Review once by yourself and remove redundant lines

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+indent_style = space
+indent_size = 2
+max_line_length = 80
+
+[*.html]
+indent_style = space
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (aashutoshrathi) [inline .editorconfig]: Why are we not trimming trailing whitespaces here?

Comment (nlok5923) [reply .editorconfig]: Sir i ventured through some of the project to know the indentation uniformity among code in most of them i find `trim_trailing_whitespace = false`

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+[*.{js,css}]
+indent_size = 2
+max_line_length = 80
+
+[*.html]
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = true
```
Comment (aashutoshrathi) [inline .editorconfig]: Do we need this line?

Comment (nlok5923) [reply .editorconfig]: i think we do not need this as it was already in `[*]`

**Review by aashutoshrathi — APPROVED** [review]

**Review by theyashshahs — APPROVED** [review]

**Pull request merged**